core/src/main/scala/io/finch/request/package.scala

```diff
@@ -261,7 +271,20 @@ package object request extends LowPriorityImplicits {
      *
      * @return an `Option` that contains a param value or `None` if the param is empty
      */
-    def apply(param: String): RequestReader[Option[String]] = RequestReader(ParamItem(param))(_.params.get(param))
+    def apply(param: String): RequestReader[Option[String]] =
+      RequestReader(ParamItem(param)){ req =>
+        // Finagle doesn't handle multipart parameter by default, so
+        // we need to add it here
+        if (req.method == Method.Post && HttpPostRequestDecoder.isMultipart(from(req))) {
```

Comment:
What do you think about making query-string params a default case here? In 99% cases we will end up fetching the query-string/url-ecoded param but with this implementation we always check whether or not it's multipart. I would simplify this:

```scala
RequestReader(ParamItem(param)) { req =>
  req.params.get(param).orElse(
    val decoder = new HttpPostRequestDecoder(from(req)) 
    decoder.getBodyHttpDatas.asScala.find(_.getName == param) flatMap {
      case attr: Attribute => Some(attr.getValue)
      case _ => None
    }
  )
}
```

I can make these changes while rebasing the PR if you don't mind.